### PR TITLE
[Feature][Connector V2] add gitlab source connector

### PR DESCRIPTION
**Pull request by liugddx**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by liugddx** [comment]
#3018

**Comment by liugddx** [comment]
test result .
image

**Comment by liugddx** [comment]
```
env {
  # You can set flink configuration here
  execution.parallelism = 1
  job.mode = "STREAMING"
  #execution.checkpoint.interval = 10000
  #execution.checkpoint.data-uri = "hdfs://localhost:9000/checkpoint"
}

source {
  # This is a example source plugin **only for test and demonstrate the feature source plugin**
  Gitlab{
    url = "https://gitlab.com/api/v4/projects"
    access_token = "xxxx"
    schema {
      fields {
        id = int
        description = string
        name = string
        name_with_namespace = string
        path = string
        http_url_to_repo = string
      }
    }
  }

  # If you would like to get more information about how to configure seatunnel and see full list of source plugins,
  # please go to https://seatunnel.apache.org/docs/category/source-v2
}

transform {


  # If you would like to get more information about how to configure seatunnel and see full list of transform plugins,
  # please go to https://seatunnel.apache.org/docs/category/transform
}

sink {
  Console {
    parallelism = 1
  }

  # If you would like to get more information about how to configure seatunnel and see full list of sink plugins,
  # please go to https://seatunnel.apache.org/docs/category/sink-v2
}
```

**Comment by liugddx** [comment]
@TaoZex @EricJoy2048 @TyrantLucifer @Hisoka-X  PTAL thanks.

**Comment by TaoZex** [comment]
Please Add e2e test.

**Comment by liugddx** [comment]
> Please Add e2e test.
It contains sensitive information and cannot be disclosed.

**Comment by TaoZex** [comment]
reference
https://github.com/apache/incubator-seatunnel/pull/3340

mock http server to add e2e test

**Comment by liugddx** [comment]
> reference #3340
> 
> mock http server to add e2e test

this mock test has covered the http use case.This connector is only for gitlab

**Comment by liugddx** [comment]
The test of restful api focuses on request rules, not result parsing.Can we store keys in `Actions secrets` and replace them with config?
https://docs.github.com/cn/actions/security-guides/security-hardening-for-github-actions#overview

@TaoZex @EricJoy2048 @hailin0

**Comment by liugddx** [comment]
> The test of restful api focuses on request rules, not result parsing.Can we store keys in `Actions secrets` and replace them with config? https://docs.github.com/cn/actions/security-guides/security-hardening-for-github-actions#overview
> 
> @TaoZex @EricJoy2048 @hailin0

@ashulin @Hisoka-X @CalvinKirs @TyrantLucifer  Please give us some advice.

**Comment by liugddx** [comment]
@hailin0 @EricJoy2048 PTAL

**Review by hailin0 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-http/connector-http-gitlab/src/main/java/org/apache/seatunnel/connectors/seatunnel/gitlab/source/GitlabSource.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.apache.seatunnel.connectors.seatunnel.http.source.HttpSourceReader;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import com.google.auto.service.AutoService;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@AutoService(SeaTunnelSource.class)
+public class GitlabSource extends HttpSource {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-http/connector-http-gitlab/src/main/java/org/apache/seatunnel/connectors/seatunnel/gitlab/source/GitlabSource.java:42]: add this method
```java
    @Override
    public Boundedness getBoundedness() {
        if (JobMode.BATCH.equals(jobContext.getJobMode())) {
            return Boundedness.BOUNDED;
        }
        throw new UnsupportedOperationException(xxx);
    }
```

**Review by hailin0 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**